### PR TITLE
[move source language] Fix Handling of Unreachable Code

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Removed overly clever code in HLIR translate to deal with unreachable code
  - This code was brittle and buggy
- All unreachable checks are now done with the CFG

## Test Plan

- added @tzakian's bug as a test case
- cargo test

### CODE REVIEW

File: language/move-lang/src/cfgir/cfg.rs [hunk 1]
```diff
@@ -140,6 +139,64 @@ impl<'a> CFG for BlockCFG<'a> {
     }
 }
 
+const DEAD_ERR_CMD: &str =
+    "Unreachable code. This statement (and any following statements) will \
+     not be executed. In some cases, this will result in unused resource values.";
+
+const DEAD_ERR_EXP: &str = "Invalid use of a divergent expression. The code following the \
+evaluation of this expression will be dead and should be removed. In some cases, this is necessary \
+to prevent unused resource values.";
```
Comment (tzakian) [inline language/move-lang/src/cfgir/cfg.rs]: Rust formatting between these two seem inconsistent. Any way to fix?

File: language/move-lang/tests/move_check/liveness/loop_weirdness.exp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+error: 
+
+   ┌── tests/move_check/liveness/loop_weirdness.move:9:36 ───
+   │
+ 9 │                 if (my_local >= 0) { break; };
+   │                                    ^^^^^^^^^^ Unreachable code. This statement (and any following statements) will not be executed. In some cases, this will result in unused resource values.
```
Comment (tzakian) [inline language/move-lang/tests/move_check/liveness/loop_weirdness.exp]: Only question I have is with the span of this error reporting; shouldn't it be only on the part including/after the `;` and not on the break itself? But I'm not even fully convinced that's a good idea, more just something to think about.

File: language/move-lang/src/cfgir/cfg.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/cfgir/cfg.rs]: I'll play around with the `\`

File: language/move-lang/tests/move_check/liveness/loop_weirdness.exp (hunk 2 above)
Comment (tnowacki) [reply language/move-lang/tests/move_check/liveness/loop_weirdness.exp]: It should point to just the `;`. 

The issue is we don't track the location of the last semicolon currently, which makes the location of the `()` inserted inaccurate. 

@bob-wilson, we should fix this

Comment (bob-wilson) [reply language/move-lang/tests/move_check/liveness/loop_weirdness.exp]: Is there an AST to attach the span for that? It's a Sequence that ends with a boxed optional expression, and in this case the expression is missing and the optional is None. Do you want to change the AST to something different?

Comment (bob-wilson) [reply language/move-lang/tests/move_check/liveness/loop_weirdness.exp]: I discussed this with Todd. I'll add a new field to the AST to record the semicolon location.

**Pull request closed without merging**